### ISSUE TITLE
Deleting projects as a cluster member deletes all the namespaces without an option to unselect the namespaces

### ISSUE THREAD

**Issue by anupama2501** [rancher/dashboard#9517]
Setup

Rancher version: 2.7-head `68bd3be`
Browser type & version: Chrome

**Describe the bug**
When a cluster member creates a project and includes a namespace in it, the deletion of the project also results in the removal of the namespaces without the ability to opt for not deleting them. This differs with the behavior of a cluster owner. In the case of a cluster owner attempting to delete a project, users are given an option to whether the namespaces should be removed or not. 

**To Reproduce**
1. Create a downstream cluster of any type, I tried with k3s/rke2 node driver clusters
2. From users and auth >> users >> create a standard user - user1.
3. Add user1 to the cluster as a cluster member.
4. Login as user1
5. Create a project P1 and a namespace ns1 from the UI 

**Result**
Following prompt is given to the users and the namespaces are deleted from the cluster.
```
  You are attempting to delete the Project Cubos. This will
       also delete all namespaces in the project:

       cubos.
```

**Expected Result**
Expected the option to keep the namespaces even when a project is deleted.  Expected the behavior to be same for cluster owner and cluster member.

**Screenshots**
Cluster member:
<img width="433" alt="2023-08-09_09-40-50" src="https://github.com/rancher/dashboard/assets/38144301/51fd13d1-b6e1-40fa-a1b0-73381384beaf">

Cluster Owner:
<img width="416" alt="2023-08-09_10-07-27" src="https://github.com/rancher/dashboard/assets/38144301/928990c0-2c3f-4419-a67a-69bce0493d1b">

**Additional context**

- From the API, when we try to delete the project, the namespaces are not deleted and they go to the `Not in a project` list.
- The behavior is seen from v2.7.0
- Not seen in v2.6.7 but seen from v2.6.9

**Comment by richard-cox** [comment]
Moving out of v2.7q3, given we've passed code freeze

**Comment by gaktive** [comment]
Will triage as part of what's after Q3 since it's not a good behaviour but this has been around a while.

**Comment by anupama2501** [comment]
@aalves08 re opening the ticket, I see this behavior on v2.8-head `6376016 `

- Create a downstream cluster of any type, I tried with k3s/rke2 node driver clusters
- Enabled auth provider
- Added a user from auth provider to the cluster as a cluster member.
- Login as that user
- Create a project P1 and a 2 namespaces ns1 and ns2
- When we try to delete the project by selecting the delete option for the namespaces, project appears back in the UI





https://github.com/rancher/dashboard/assets/38144301/33b2563f-7658-4fe4-992d-d113c3bdaf12

**Comment by aalves08** [comment]
Hello @anupama2501 . Since we didn't have any more updates since friday, we need to bump this out of `2.8.0` as it's not a blocking issue. It's just that this view doesn't auto-update and the user needs to refresh the page OR navigate away to get the view updated (at least for the the lingering deleted project).

FYI @nwmac @gaktive @richard-cox

**Comment by aalves08** [comment]
Some comments from @Sean-McQ , who also looked into this in an effort to get this fixed before code freeze (so that we don't lose context):

> I wasn't able to reproduce the project header staying in the list but I was able to reproduce the namespaces continuing to show in `terminating` status without disappearing.

> It looks like theres a race condition wherein the roleTemplates and roleTemplateBindings are removed for a namespace before a seperate process (presumably checking the namespace for potentially orphaned resources) has to complete before the namespace can actually be removed. Since projects are effectively namespaces on the local cluster, this applies to that as well.

> So basically, the user in question, if they only have project permissions to a project, they will lose permissions to see further updates indicating the deleted resources before they can actually see the deletion of the resources involved which means that...

> A user deletes a resource and the store is updated via the VUEX transaction 
> A WS event comes in stating that the resource is updated in some way (probably a status change to terminating) thus recreating the resource depending on the performance options set for rancher 
> the roleTemplateBindings for the resource are removed and the user can see no further updates.

> The best solution I see here is to add a secondary event after deleting a namespace or a project on this list to force a refetch on both the project and namespace lists that create the project/namespace list but I didn't have time to implement that.

**Comment by anupama2501** [comment]
Hi @aalves08, since we already merged the code for the issue would it be possible to revert the merged code since this is not going to make in 2.8.0?

**Comment by aalves08** [comment]
Did a bit of investigation related to this issue and filed a new issue on `rancher/rancher` as I believe this is related to backend: https://github.com/rancher/rancher/issues/43799

Will move this to the `backend-blocked` column, for now until more information is provided from the backend team.

FYI @gaktive @nwmac @richard-cox @anupama2501

**Comment by gaktive** [comment]
Related ticket in backend closed as won't fix/working as designed. 

Per https://github.com/rancher/rancher/issues/43799#issuecomment-2452065077, some behaviour improvements have come in that make this work better but additional work would require an RFE.
